worlds/blasphemous/Preprocessor.py

```diff
@@ -0,0 +1,579 @@
+# Preprocessor to convert Blasphemous Randomizer logic into a StringWorldDefinition for use with APHKLogicExtractor
+# https://github.com/BrandenEK/Blasphemous.Randomizer
+# https://github.com/ArchipelagoMW-HollowKnight/APHKLogicExtractor
+
+
+import json, requests, argparse
+from typing import List, Dict, Any
+
+
+def load_resource_local(file: str) -> List[Dict[str, Any]]:
+    print(f"Reading from {file}")
+    loaded = []
+    with open(file, encoding="utf-8") as f:
+        loaded = read_json(f.readlines())
+        f.close()
+
+    return loaded
+
+
+def load_resource_from_web(url: str) -> List[Dict[str, Any]]:
+    req = requests.get(url, timeout=1)
+    print(f"Reading from {url}")
+    req.encoding = "utf-8"
+    lines: List[str] = []
+    for line in req.text.splitlines():
+        while "\t" in line:
+            line = line[1::]
+        if line != "":
+            lines.append(line)
+    return read_json(lines)
+
+
+def read_json(lines: List[str]) -> List[Dict[str, Any]]:
+    loaded = []
+    creating_object: bool = False
+    obj: str = ""
+    for line in lines:
+        stripped = line.strip()
+        if "{" in stripped:
+            creating_object = True
+            obj += stripped
+            continue
+        elif "}," in stripped:
+            creating_object = False
+            obj += "}"
+            #print(f"obj = {obj}")
+            loaded.append(json.loads(obj))
+            obj = ""
+            continue
+
+        if not creating_object:
+            continue
+        else:
+            if "}," in lines[lines.index(line)+1] and stripped[-1] == ",":
+                obj += stripped[:-1]
+            else:
+                obj += stripped
+
+    return loaded
+
+
+def get_room_from_door(door: str) -> str:
+    return door[:door.find("[")]
+
+
+def preprocess_logic(is_door: bool, id: str, logic: str) -> str:
+    if id in logic and not is_door:
+        index: int = logic.find(id)
+        logic = logic[:index] + logic[index+len(id)+4:]
+
+    while ">=" in logic:
+        index: int = logic.find(">=")
+        logic = logic[:index-1] + logic[index+3:]
```

Comment:
This manner of processing seems pretty brittle to me (iiuc the goal is to convert tokens of the form `thing >= X` to `thingX` which I guess is a fine approach but would fall apart if X was 2 digits of if there was additional or less whitespace). For HK I was planning on keeping the comparison in the term and during load rules, splitting on `>` and using the split term during rule building. e.g. something like `FIREBALL>1` gets converted during world setup to `state.has("FIREBALL", 2)` instead of making it a pure dictionary lookup

Reply:
It works fine if X is 2 digits actually, and is already doing that. eg `bones >= 44` -> `bones44`

You are right that it would fall apart if there was more or less whitespace though, but the logic strings are consistent enough for that not to be a problem for me.

Reply:
DNF? Oh man, is this script just what Witness does, except it does it once statically to generate a set of python files instead of doing it at runtime? Man, I should probably look at this PR then, I'll probably understand a lot of what it's doing

Reply:
Just to clarify this point, there will never be more or less than 1 character of whitespace in between tokens, since the base randomizer will throw an error in that case and fail the logic tests.  So you can safely assume that will never be the case.